tools/internal_ci/linux/grpc_python_bazel_test_in_docker.sh

```diff
@@ -14,18 +14,23 @@
 # limitations under the License.
 #
 # Test full Bazel
-#
-# NOTE: No empty lines should appear in this file before igncr is set!
-set -ex -o igncr || set -ex
+
+set -ex
 
 mkdir -p /var/local/git
 git clone /var/local/jenkins/grpc /var/local/git/grpc
 (cd /var/local/jenkins/grpc/ && git submodule foreach 'cd /var/local/git/grpc \
 && git submodule update --init --reference /var/local/jenkins/grpc/${name} \
 ${name}')
-cd /var/local/git/grpc/test
+cd /var/local/git/grpc
 TEST_TARGETS="//src/python/... //tools/distrib/python/grpcio_tools/... //examples/python/..."
 BAZEL_FLAGS="--test_output=errors"
-bazel test ${BAZEL_FLAGS} ${TEST_TARGETS}
-bazel test --config=python_single_threaded_unary_stream ${BAZEL_FLAGS} ${TEST_TARGETS}
-bazel test --config=python_poller_engine ${BAZEL_FLAGS} ${TEST_TARGETS}
+
+python3 tools/run_tests/python_utils/bazel_report_helper.py --report_path python_bazel_tests
+python_bazel_tests/bazel_wrapper test ${BAZEL_FLAGS} ${TEST_TARGETS}
+
+python3 tools/run_tests/python_utils/bazel_report_helper.py --report_path python_bazel_tests_single_threaded_unary_streams
+python_bazel_tests_single_threaded_unary_streams/bazel_wrapper test --config=python_single_threaded_unary_stream ${BAZEL_FLAGS} ${TEST_TARGETS}
+
+python3 tools/run_tests/python_utils/bazel_report_helper.py --report_path python_bazel_tests_poller_engine
+python_bazel_tests_poller_engine/bazel_wrapper test --config=python_poller_engine ${BAZEL_FLAGS} ${TEST_TARGETS}
```

Comment:
Since the Python script will geneate the wrapper command, have we considered the alternative of sending all arguments to the Python script? For example:

```bash
python3 tools/run_tests/python_utils/bazel_report_helper.py \
   --report_path python_bazel_tests_poller_engine \
   --bazel_args="test --config=python_poller_engine ${BAZEL_FLAGS} ${TEST_TARGETS}"
```

Reply:
> Could you integrate this into the `tools/bazel` script and condition activation of the test reports on an environment variable so that we don't have to update all of the call sites?

I'd rather not, at least not initially.  I don't want to create one complex wrapper that "does everything". Currently `tools/bazel` takes care of selecting the right bazel version but doesn't touch the bazel flags at all and it doesn't know anything about CI, kokoro or resultstore, reports. Purpose of the new wrapper is completely different so I think layering purpose specific wrappers on top of each other is actually better solution that having everything together in a single wrapper.

I think having to update the call sites is actually what we want initially - it makes the introduction of the reports/uploading logic more apparent and clearly differentiates between places where we do upload structured results and where we don't (yet).

Reply:
- I think this would only make sense if python would invoke the bazel (otherwise I don't see how that simplifies stuff, we would just need to persist all the arguments into the wrapper and then use them). But invoking bazel from python is something we'd have to test (especially on windows, it could mess with things in unexpected ways).
- I think having the arbitrary args pass through python is potentially risky (we would need to pay close attention to quoting, splitting the args etc.) - I think this is extra trouble we don't need. Just forwarding "$@" in the wrapper seems safer, simpler and less error-prone.